source/extensions/tracers/opentelemetry/http_trace_exporter.cc

```diff
@@ -50,6 +50,10 @@ bool OpenTelemetryHttpTraceExporter::log(const ExportTraceServiceRequest& reques
   message->headers().setReferenceMethod(Http::Headers::get().MethodValues.Post);
   message->headers().setReferenceContentType(Http::Headers::get().ContentTypeValues.Protobuf);
 
+  // User-Agent header follows the OTLP specification:
+  // https://github.com/open-telemetry/opentelemetry-specification/blob/v1.30.0/specification/protocol/exporter.md#user-agent
+  message->headers().setReferenceUserAgent("OTel-OTLP-Exporter-Envoy");
```

Comment:
The spec also mentions to add a version to the header. Is there a way to obtain the Envoy version here? cc @wbpcode @htuch

Reply:
Yes, you can get this from `source/common/version/version.h`.

Reply:
Alright, thanks for the pointer! I have added it now.